### DISCUSSION TITLE
Same build for unit tests and XCUITest

### DISCUSSION THREAD

**Discussion by ChrisSmartNews** [fastlane/fastlane#20426] [Q&A]
Hello, is it possible to say have a `run_tests` for unit tests and a `run_tests` for UI tests, and let them share the same build so they don't have to compile the app twice?

Currently I have them running in two different jobs on circleCI and both take a long time ~15 minutes to compile and build.

**Comment by snelloru** [comment]
Thanks for asking this. I also found that we do build twice in our pipelines, once for actual build and another one during the tests. I wonder why can't we can utilize the IPA file built already in another job. Anyone doing this already?

**Comment by Carissa34** [comment]
Thanks for sharing. I found a lot of interesting information here. [MyCCPay Login](https://www.myccpay.me/)

**Comment by doriand-itx** [comment]
Hello, any updates ? I'm really interested too. Thanks.
